### ISSUE TITLE
Add GHA for examples coverage?

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1675]
TIL covr supports coverage for examples. This will be good to get a sense of where there's major gaps in what's in examples.

https://covr.r-lib.org/reference/package_coverage.html

Has anyone seen a package using this feature in GHA? does codecov support something like having two coverage layers? or maybe we need to do something like gh-pages where we have one branch used just to calculate examples coverage?

**Comment by IndrajeetPatil** [comment]
Working on this in #1785.

- Should this check fail if coverage is less than 100%?
- Should all files be included, or only the files containing linters?

---

For a context, the current code coverage of files via examples looks like this:

```r
R/comments.R: 0.00%
R/deprecated.R: 0.00%
R/lintr-deprecated.R: 0.00%
R/use_lintr.R: 0.00%
R/extract.R: 11.59%
R/cache.R: 14.55%
R/methods.R: [22](https://github.com/r-lib/lintr/actions/runs/3595761726/jobs/6055644751#step:5:23).76%
R/lint.R: 24.02%
R/exclude.R: 36.60%
R/get_source_expressions.R: 41.67%
R/namespace.R: 46.67%
R/unnecessary_lambda_linter.R: 56.52%
R/make_linter_from_regex.R: 57.41%
R/with.R: 62.50%
R/settings.R: 65.31%
R/with_id.R: 66.67%
R/expect_lint.R: 69.33%
R/nonportable_path_linter.R: 71.43%
R/backport_linter.R: 72.58%
R/trailing_blank_lines_linter.R: 72.73%
R/utils.R: 72.87%
R/linter_tags.R: 73.17%
R/object_usage_linter.R: 77.10%
R/ids_with_token.R: 77.78%
R/namespace_linter.R: 82.35%
R/settings_utils.R: 83.61%
R/tree-utils.R: 86.67%
R/xml_nodes_to_lints.R: 90.24%
R/path_utils.R: 90.32%
R/literal_coercion_linter.R: 90.57%
R/actions.R: 90.91%
R/undesirable_operator_linter.R: 91.89%
R/object_name_linter.R: 93.44%
R/indentation_linter.R: 94.51%
R/infix_spaces_linter.R: 95.35%
R/fixed_regex_linter.R: 96.51%
R/semicolon_linter.R: 96.77%
R/unused_import_linter.R: 97.14%
R/sprintf_linter.R: 97.67%
R/comment_linters.R: 98.44%
R/absolute_path_linter.R: 100.00%
R/any_duplicated_linter.R: 100.00%
R/any_is_na_linter.R: 100.00%
R/assignment_linter.R: 100.00%
R/boolean_arithmetic_linter.R: 100.00%
R/brace_linter.R: 100.00%
R/class_equals_linter.R: 100.00%
R/commas_linter.R: 100.00%
R/condition_message_linter.R: 100.00%
R/conjunct_test_linter.R: 100.00%
R/consecutive_stopifnot_linter.R: 100.00%
R/cyclocomp_linter.R: 100.00%
R/declared_functions.R: 100.00%
R/duplicate_argument_linter.R: 100.00%
R/empty_assignment_linter.R: 100.00%
R/equals_na_linter.R: 100.00%
R/expect_comparison_linter.R: 100.00%
R/expect_identical_linter.R: 100.00%
R/expect_length_linter.R: 100.00%
R/expect_named_linter.R: 100.00%
R/expect_not_linter.R: 100.00%
R/expect_null_linter.R: 100.00%
R/expect_s3_class_linter.R: 100.00%
R/expect_s4_class_linter.R: 100.00%
R/expect_true_false_linter.R: 100.00%
R/expect_type_linter.R: 100.00%
R/extraction_operator_linter.R: 100.00%
R/for_loop_index_linter.R: 100.00%
R/function_argument_linter.R: 100.00%
R/function_left_parentheses_linter.R: 100.00%
R/function_return_linter.R: 100.00%
R/ifelse_censor_linter.R: 100.00%
R/implicit_integer_linter.R: 100.00%
R/inner_combine_linter.R: 100.00%
R/is_lint_level.R: 100.00%
R/is_numeric_linter.R: 100.00%
R/lengths_linter.R: 100.00%
R/line_length_linter.R: 100.00%
R/missing_argument_linter.R: 100.00%
R/missing_package_linter.R: 100.00%
R/nested_ifelse_linter.R: 100.00%
R/numeric_leading_zero_linter.R: 100.00%
R/object_length_linter.R: [100](https://github.com/r-lib/lintr/actions/runs/3595761726/jobs/6055644751#step:5:101).00%
R/outer_negation_linter.R: 100.00%
R/package_hooks_linter.R: 100.00%
R/paren_body_linter.R: 100.00%
R/paste_linter.R: 100.00%
R/pipe_call_linter.R: 100.00%
R/pipe_continuation_linter.R: 100.00%
R/redundant_equals_linter.R: 100.00%
R/redundant_ifelse_linter.R: 100.00%
R/regex_subset_linter.R: 100.00%
R/routine_registration_linter.R: 100.00%
R/seq_linter.R: 100.00%
R/single_quotes_linter.R: 100.00%
R/spaces_inside_linter.R: 100.00%
R/spaces_left_parentheses_linter.R: 100.00%
R/string_boundary_linter.R: 100.00%
R/strings_as_factors_linter.R: 100.00%
R/system_file_linter.R: 100.00%
R/T_and_F_symbol_linter.R: 100.00%
R/trailing_whitespace_linter.R: 100.00%
R/undesirable_function_linter.R: 100.00%
R/unnecessary_placeholder_linter.R: 100.00%
R/unneeded_concatenation_linter.R: 100.00%
R/unreachable_code_linter.R: 100.00%
R/vector_logic_linter.R: 100.00%
R/xp_utils.R: 100.00%
R/yoda_test_linter.R: 100.00%
```

**Comment by IndrajeetPatil** [comment]
How about we restrict this only to `*_linter.R` files and set the threshold to 90%? That is, the builds will fail if any of these files have example coverage of less than 90%. 

Why not 100%? Well, because we don't need/want to have examples showcasing every exception we are handling.

**Comment by MichaelChirico** [comment]
I'm not sure about the restriction to `*_linter.R` files, e.g. these seem useful to cover:

```
R/settings_utils.R: 83.61%
R/tree-utils.R: 86.67%
```

How about an exclusion list instead? e.g. R/deprecated.R clearly no need to cover.

90% is fine.
